examples/medplum-photon-integration/src/bots/sync-patient-from-photon.ts

```diff
@@ -0,0 +1,389 @@
+          return {
+            fullUrl: 'urn:uuid:' + randomUUID(),
+            request: { method: 'POST', url: 'AllergyIntolerance' },
+            resource: allergy,
+          };
+        });
+        batch.entry?.push(...allergyEntries);
+      }
+
+      // If there are prescriptions, create entries and add them to the bundle
+      if (prescriptions) {
+        const prescriptionEntries: BundleEntry[] = prescriptions.map((prescription) => {
+          return {
+            fullUrl: 'urn:uuid:' + randomUUID(),
+            request: { method: 'POST', url: 'MedicationRequest' },
+            resource: prescription,
+          };
+        });
+        batch.entry?.push(...prescriptionEntries);
+      }
+    }
+  }
+
+  // execute the bundle
+  await medplum.executeBatch(batch);
+}
+
+export async function checkForExistingPatient(photonPatient: PhotonPatient, medplum: MedplumClient): Promise<boolean> {
+  let patient: Patient | undefined;
+  try {
+    patient = await medplum.searchOne('Patient', {
+      identifier: NEUTRON_HEALTH_PATIENTS + '|' + photonPatient.id,
+    });
+    if (patient) {
+      return true;
+    }
+  } catch (err) {
+    console.error(`Error for Patient ID ${photonPatient.id}:`, normalizeErrorString(err));
+  }
+
+  if (photonPatient.externalId) {
+    try {
+      patient = await medplum.readResource('Patient', photonPatient.externalId);
+      if (patient) {
+        return true;
+      }
+    } catch (err) {
+      console.error(`Error for Patient ID ${photonPatient.id}:`, normalizeErrorString(err));
+    }
+  }
+
+  return false;
+}
+
+export function createPatientResource(photonPatient: PhotonPatient): Patient {
+  const telecom: ContactPoint[] = [{ system: 'phone', value: photonPatient.phone }];
+  if (photonPatient.email) {
+    telecom.push({ system: 'email', value: photonPatient.email });
+  }
+
+  const patient: Patient = {
+    resourceType: 'Patient',
+    identifier: [{ system: NEUTRON_HEALTH_PATIENTS, value: photonPatient.id }],
+    name: [{ family: photonPatient.name.last, given: [photonPatient.name.first] }],
+    gender: photonPatient.sex.toLowerCase() as Patient['gender'],
+    birthDate: photonPatient.dateOfBirth,
+    telecom,
+  };
+
+  const address = photonPatient.address && formatAddress(photonPatient.address);
+
+  if (address) {
+    patient.address = address;
+  }
+
+  return patient;
+}
+
+export function createAllergies(
+  patientReference: Reference<Patient>,
+  photonAllergies?: PhotonPatientAllergy[]
+): AllergyIntolerance[] | undefined {
+  if (!photonAllergies || photonAllergies.length === 0) {
+    return undefined;
+  }
+  const allergies: AllergyIntolerance[] = [];
+  for (const photonAllergy of photonAllergies) {
+    const { allergen, comment, onset } = photonAllergy;
+    if (!allergen.rxcui) {
+      continue;
+    }
+
+    const allergy: AllergyIntolerance = {
+      resourceType: 'AllergyIntolerance',
```

Comment:
We should also add `meta.source=<PHOTON_URL>` to the `AllergyIntolerance` and `MedicationRequest` . this is a searchable parameter that can be used for upsert